src/customio.cpp

```diff
@@ -0,0 +1,242 @@
+#include "customio.h"
+
+#include "entity.h"
+#include "entity/cbaseentity.h"
+#include "entity/ccsplayercontroller.h"
+#include "entity/cenventitymaker.h"
+#include "entity/cphysthruster.h"
+
+#include <baseentity.h>
+
+#include <entity/cbasetrigger.h>
+#include <string>
+#include <vector>
+
+struct AddOutputKey_t
+{
+    AddOutputKey_t(const char* pName, int32_t parts) :
+        m_pName(pName)
+    {
+        m_nLength = strlen(pName);
+        m_nParts  = parts;
+    }
+
+    AddOutputKey_t(const AddOutputKey_t& other) :
+        m_pName(other.m_pName),
+        m_nLength(other.m_nLength),
+        m_nParts(other.m_nParts) {}
+
+    const char* m_pName;
+    size_t      m_nLength;
+    int32_t     m_nParts;
+};
+
+using AddOutputHandler_t = void (*)(Z_CBaseEntity*                  pInstance,
+                                    CEntityInstance*                pActivator,
+                                    CEntityInstance*                pCaller,
+                                    const std::vector<std::string>& vecArgs);
+
+struct AddOutputInfo_t
+{
+    AddOutputInfo_t(const AddOutputKey_t& key, const AddOutputHandler_t& handler) :
+        m_Key(key), m_Handler(handler) {}
+
+    AddOutputKey_t     m_Key;
+    AddOutputHandler_t m_Handler;
+};
+
+static void AddOutputCustom_Targetname(Z_CBaseEntity*                  pInstance,
+                                       CEntityInstance*                pActivator,
+                                       CEntityInstance*                pCaller,
+                                       const std::vector<std::string>& vecArgs)
+{
+    addresses::CEntityIdentity_SetEntityName(pInstance->m_pEntity, vecArgs[1].c_str());
+    Message("SetName %s to %d", vecArgs[1].c_str(), pInstance->GetHandle().GetEntryIndex());
```

Comment:
Probably best to put these logs behind `#ifdef _DEBUG`, this applies to all the other handlers.